frontend/src/components/atoms/elements/LightboxWrapper.tsx

```diff
@@ -0,0 +1,52 @@
+import Lightbox from 'yet-another-react-lightbox';
+import Download from 'yet-another-react-lightbox/plugins/download';
+import Zoom from 'yet-another-react-lightbox/plugins/zoom';
+
+import 'yet-another-react-lightbox/styles.css';
+
+interface LightboxWrapperProps {
+  isOpen: boolean;
+  onClose: () => void;
+  imageUrl: string;
+  imageName: string;
+}
+
+const LightboxWrapper = ({
+  isOpen,
+  onClose,
+  imageUrl,
+  imageName
+}: LightboxWrapperProps) => {
+  return (
+    <Lightbox
+      open={isOpen}
+      close={onClose}
+      slides={[{ src: imageUrl }]}
+      carousel={{ finite: true }}
+      render={{ buttonPrev: () => null, buttonNext: () => null }}
+      plugins={[Zoom, Download]}
+      zoom={{
+        maxZoomPixelRatio: 5,
+        zoomInMultiplier: 2
+      }}
+      download={{
+        download: async ({ slide }) => {
+          try {
+            const response = await fetch(slide.src, { mode: 'cors' });
+            const blob = await response.blob();
+            const url = window.URL.createObjectURL(blob);
+            const link = document.createElement('a');
+            link.href = url;
+            link.download = imageName || 'image';
+            link.click();
+            window.URL.revokeObjectURL(url);
```

Comment:
Why do we have a custom `fetch()` here rather than just `download=true`?

Download URL's do not require auth tokens, nor will it in the future (we're using cookie auth instead #1521).

Ref: https://yet-another-react-lightbox.com/plugins/download

Reply:
I had this idea of keeping plugins to a minimum, but with the lazy load you suggested, that might not be needed anymore. 
And, to be honest, this is my first try on TS ever, so iv been confused.

Spent way to long trying to make E2E tests, before I had to put everything on hold.

Reply:
Pretty good for a first try. :)

I know, building these tests, especially in the beginning, is quite tedious. But once you get the hang of it, additional tests (per project), go faster. Really appreciate your work on this, thanks!

Now on this issue, you're already using the download plugin. ;)

Just putting the URL there might work. Let me know if you're gonna add that to the PR. With that, I'm happy to merge.

Reply:
Ill look into it after work tomorrow and report back. Thanks for the guidance and for all the amazing work you and your team does <3